Content.Client/Doors/DoorSystem.cs

```diff
@@ -18,28 +21,23 @@ public override void Initialize()
         {
             base.Initialize();
 
-            SubscribeLocalEvent<DoorStateMessage>(HandleDoorState);
+            SubscribeLocalEvent<ClientDoorComponent, DoorStateChangedEvent>(OnDoorStateChanged);
         }
 
-        /// <summary>
-        /// Registers doors to be periodically checked.
-        /// </summary>
-        /// <param name="message">A message corresponding to the component under consideration, raised when its state changes.</param>
-        private void HandleDoorState(DoorStateMessage message)
+        private void OnDoorStateChanged(EntityUid uid, ClientDoorComponent door, DoorStateChangedEvent args)
         {
-            switch (message.State)
-            {
-                case SharedDoorComponent.DoorState.Closed:
-                case SharedDoorComponent.DoorState.Open:
-                    _activeDoors.Remove(message.Component);
-                    break;
-                case SharedDoorComponent.DoorState.Closing:
-                case SharedDoorComponent.DoorState.Opening:
-                    _activeDoors.Add(message.Component);
-                    break;
-                default:
-                    throw new ArgumentOutOfRangeException();
-            }
+            if (args.State == DoorState.Open)
+                _activeDoors.Remove(door);
+            else if (args.State == DoorState.Opening)
+                _activeDoors.Add(door);
```

Comment:
What about the closed/closing state? Don't quite see why the switch-case had to go, could have kept it in and just continued the new code after?

Reply:
Ah yep I just made a mistake here. I probably should have just copy & pasted the old code over. Thanks for spotting that.